### PR TITLE
feat(transformer): try another way in new target plugin

### PR DESCRIPTION
**Pull request by Dunqing**
Try https://github.com/oxc-project/oxc/pull/1967#discussion_r1447364495

### CODE REVIEW

**Comment by Dunqing** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/1991?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `main`
  * **PR #1990** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/1990?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #1991** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/1991?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/oxc-project/oxc/1991?utm_source=stack-comment).

**Comment by Dunqing** [comment]
Do you think this is a viable solution? @Boshen

File: crates/oxc_transformer/src/es2015/new_target.rs [hunk 1]
```diff
@@ -84,6 +70,10 @@ impl<'a> NewTarget<'a> {
     }
 
     pub fn transform_expression<'b>(&mut self, expr: &'b mut Expression<'a>) {
+        // get last kind
+        let kind = self.ctx.nodes().kind(*self.ctx.current_node_id.borrow());
+
+        // If we change the expression, we need to update semantic info
```
Comment (Dunqing) [inline crates/oxc_transformer/src/es2015/new_target.rs:76]: we need to consider how to go about doing this

**Comment by Boshen** [comment]
I think we need to implement a new stack for the transformer.

`AstNodes` is append only and stores all the ast nodes, which is not ideal for the transformer where it needs to remove ast nodes.

**Comment by Dunqing** [comment]
remove import if only have type reference

> I think we need to implement a new stack for the transformer.
> 
> `AstNodes` is append only and stores all the ast nodes, which is not ideal for the transformer where it needs to remove ast nodes.

So scopes / symbols also need to do same thing?

**Pull request closed without merging**